Mathlib/Geometry/Euclidean/Angle/Unoriented/TriangleInequality.lean

```diff
@@ -0,0 +1,220 @@
+  rw [inner_product_of_units_as_cos Hx Hy]
+  exact Real.abs_cos_le_one (angle x y)
+
+lemma sin_as_inner_product {x y : V} (Hx: ‖x‖ = 1) (Hy: ‖y‖ = 1) :
+    let u := unit (x - ⟪x, y⟫ • y)
+    ⟪x, u⟫ = Real.sin (angle x y) := by
+  intro u; unfold u
+  have H0 := angle_triangle_aux1 Hx Hy; simp only at H0
+  nth_rw 2 [inner_product_of_units_as_cos Hx Hy] at H0
+  rw [Real.cos_sq'] at H0
+  have H0 : Real.sin (angle x y) ^ 2 = ⟪x, unit (x - ⟪x, y⟫ • y)⟫ ^ 2 := by
+    linarith
+  rw [sq_eq_sq_iff_abs_eq_abs] at H0
+  rw [abs_of_nonneg (inner_product_with_proj_nonneg Hx Hy)] at H0
+  rw [abs_of_nonneg (sin_angle_nonneg x y)] at H0
+  exact H0.symm
+
+lemma eq_of_inner_eq_one {x y : V} (Hx : ‖x‖ = 1) (Hy : ‖y‖ = 1) (H : ⟪x, y⟫ = 1) :
+    x = y := by
+  rw [inner_product_of_units_as_cos Hx Hy] at H
+  rw [cos_eq_one_iff_angle_eq_zero] at H
+  rw [angle_eq_zero_iff] at H
+  obtain ⟨H, r, H0, H1⟩ := H
+  rw [H1, norm_smul, Hx] at Hy
+  simp only [Real.norm_eq_abs, mul_one] at Hy
+  rw [abs_of_pos H0] at Hy
+  rw [Hy] at H1; simp only [one_smul] at H1
+  exact H1.symm
+
+lemma eq_neg_of_inner_eq_neg_one {x y : V} (Hx : ‖x‖ = 1) (Hy : ‖y‖ = 1)
+    (H : ⟪x, y⟫ = -1) :
+    x = -y := by
+  rw [inner_product_of_units_as_cos Hx Hy] at H
+  rw [cos_eq_neg_one_iff_angle_eq_pi] at H
+  rw [angle_eq_pi_iff] at H
+  obtain ⟨H, r, H0, H1⟩ := H
+  rw [H1, norm_smul, Hx] at Hy;
+  simp only [Real.norm_eq_abs, mul_one] at Hy
+  rw [abs_of_neg H0, neg_eq_iff_eq_neg] at Hy
+  rw [Hy] at H1; simp only [neg_smul, one_smul] at H1; rw [H1]; simp
+
+lemma angle_triangle_aux2 {x y : V} (Hx : ‖x‖ = 1) (Hy : ‖y‖ = 1) :
+    let u := unit (x - ⟪x, y⟫ • y)
+    x = Real.cos (angle x y) • y + Real.sin (angle x y) • u := by
+  rw [← sin_as_inner_product Hx Hy]
+  dsimp [unit]
+  rw [← inner_product_of_units_as_cos Hx Hy, real_inner_smul_right]
+  rw [inner_sub_right, real_inner_smul_right]
+  rw [inner_sq_of_unit Hx, ← sq]
+  rw [← smul_assoc]
+  obtain H | H := em (‖x - ⟪x, y⟫ • y‖ = 0)
+  · simp only [H, inv_zero, zero_mul, smul_eq_mul, mul_zero, zero_smul, add_zero]
+    rwa [norm_eq_zero, sub_eq_zero] at H
+  · have H1 : ⟪x, y⟫ ≠ 1 := by
+      intro H2; apply H; simp only [norm_eq_zero]; rw [H2]
+      simp only [one_smul]; rw [sub_eq_zero]
+      apply eq_of_inner_eq_one Hx Hy H2
+    have H2 : ⟪x, y⟫ ≠ -1 := by
+      intro H3; apply H; simp only [norm_eq_zero];
+      simp only [H3, neg_smul, one_smul, sub_neg_eq_add, add_eq_zero_iff_eq_neg]
+      apply eq_neg_of_inner_eq_neg_one Hx Hy H3
+    have H3 : 1 - ⟪x, y⟫ ^ 2 ≠ 0 := by
+      rw [sub_ne_zero, ne_comm, sq_ne_one_iff]
+      tauto
+    field_simp [← sq, ← real_inner_self_eq_norm_sq,
+      inner_sub_left, inner_sub_right, inner_sub_right,
+      real_inner_smul_left, real_inner_smul_right,
+      real_inner_comm x y, inner_sq_of_unit Hx, inner_sq_of_unit Hy]
+
+@[simp]
+lemma proj_ortogonal_of_unit (x : V) {y : V} (Hy : ‖y‖ = 1):
+    let u := unit (x - ⟪x, y⟫ • y)
+    ⟪y, u⟫ = 0 := by
```

Comment:
There's a typo `ortogonal` in this name, but the name also needs rewriting following the naming conventions, probably yielding something more like `inner_sub_inner_smul_right_eq_zero_of_norm_eq_one` (and this would go in a general inner product space file).